src/hierarchies/priors/nig_prior_model.cc

```diff
@@ -84,8 +84,7 @@ double NIGPriorModel::lpdf(const google::protobuf::Message &state_) {
   return target;
 }
 
-std::shared_ptr<google::protobuf::Message> NIGPriorModel::sample(
-    ProtoHypersPtr hier_hypers) {
+State::UniLS NIGPriorModel::sample(ProtoHypersPtr hier_hypers) {
```

Comment:
Can this also become a const ref? I think so

Reply:
const ref to a pointer?

Reply:
I didn't notice it was a pointer. Still, it could be a const pointer. But it's not like it matters much anyway